pkgs/applications/editors/emacs/elisp-packages/manual-packages/zstd/package.nix

```
@@ -0,0 +1,41 @@
+{
+  lib,
+  melpaBuild,
+  pkgs,
+  fetchFromGitHub,
+}:
+
+let
+  libExt = pkgs.stdenv.hostPlatform.extensions.sharedLibrary;
+in
+melpaBuild {
+  pname = "zstd";
+  version = "0-unstable-2020-06-03";
+
+  src = fetchFromGitHub {
+    owner = "syohex";
+    repo = "emacs-zstd";
+    rev = "072b264e2cbd5c05be06a1208ebccc2dab44be39";
+    hash = "sha256-p8bxefytTOSV6vIG8PAPBXfVKA2rfmWdRtVwjE42mAw=";
+  };
+
+  ignoreCompilationError = false;
+
+  buildInputs = [ pkgs.zstd ];
+
+  preBuild = ''
+    $CC -shared -o zstd-core${libExt} zstd-core.c -lzstd
+  '';
+
+  preInstall = ''
+    outd=$out/share/emacs/site-lisp/elpa/zstd-20200603.0
+    install -Dm755 -t $outd zstd-core${libExt}
+  '';
```

`zstd-20200603.0` can be referred to as `$ename-$melpaVersion` in bash.  However, I think it is better to add .so to the recipe.
```suggestion
  files = ''(:defaults "zstd-core${libExt}")'';
```

---

That has worked, thank you.